src/stage1/hash_map.hpp

```diff
@@ -51,7 +77,7 @@ class HashMap {
 
         if (_index_bytes == nullptr) {
             if (_entries.length < 16) {
-                _entries.append({HashFunction(key), 0, key, value});
+                _entries.append({HashFunction(MakePointer<K>::convert(key)), 0, key, value});
```

Comment:
forgive my lack of C++ knowledge but why not just `&key` ?

Reply:
MakePointer selects an implementation based on the type.  When the key is `*T`, it's `key`.  When the key is `T`, it's `&key`.

Reply:
mmm I'd feel a lot more comfortable with the callsites being updated to use `&` where appropriate. One C++ feature here, one C++ feature there, and then soon enough Bjarne Soustroup is murdering your firstborn child

Reply:
That doesn't work.  When the key is a pointer, we need to pass it by value.

Reply:
ahh I see the problem. Alright, I'm still uncomfortable with this C++ creeping in, but I think it's justified enough for me to not push back on it any more.